### DISCUSSION TITLE
Vue router has a very fool design

### DISCUSSION THREAD

**Discussion by bigradish** [vuejs/router#1931] [Show and tell]
The idea of the design of vue router is very fool.
For example: https://github.com/vuejs/router/blob/main/packages/router/CHANGELOG.md#414-2022-08-22
The designer of vue router did many things which should not be considered by him and confined the user's capability.
Hope @yyx990803 can review the design idea of vue router.
Thank you very much.

**Comment by posva** [comment]
Your blind belief without reasoning about my design decisions, despite my extensive involvement in [multiple RFCs](https://github.com/vuejs/rfcs/pulls?q=is%3Apr+sort%3Aupdated-desc+author%3Aposva+), is shallow. Defending an anti-pattern that has been advised [against and unsupported since 2016](https://github.com/vuejs/vue-router/issues/444#issuecomment-204657658) shows a lack of research on your part. Directly addressing Evan as if I need to be parented is disrespectful and unwelcome here. Your behavior only makes a fool of yourself.

**Comment by bigradish** [comment]
@posva ,
I appologize for my words that may offend you. But what I mean is the design and is not the person who make the design.
Even if the modification is reasonable, it makes the new minor version incompatible with the previous ones, so it should be put into the next big version.
Thank you.

**Reply by posva** [reply]
As said in the past, it was explicitly unsupported that's why it didn't go through a new major version.

As I pointed out, the problem is not the choice of words...
